### ISSUE TITLE
Crashlog - caused by pasting into label

### ISSUE THREAD

**Issue by roughana** [fadden/6502bench#82]
[CrashLog.txt](https://github.com/fadden/6502bench/files/4941817/CrashLog.txt)
I copied text from an operand towards the end of a $3000 length project and moved back to near the top and tried to paste it in as a label after using ^L to open the label dialog.
Windows 8

**Comment by fadden** [comment]
System.Runtime.Remoting.RemotingException: Object [...] has been disconnected or does not exist at the server.

Hmm.  This is the app-domain security sandbox timing out.  The extension scripts get notified when a label changes.

Have you fiddled with the debug features?  If so, is "Use Keep-Alive Hack" checked when you open the DEBUG menu?  (It's supposed to be.)

If you're curious: https://stackoverflow.com/q/52230527/294248

**Comment by roughana** [comment]
No, I haven't fiddled with that. Use Keep-Alive Hack is checked.
My wifi connection sometimes goes down. This laptop and my router both have issues.
Would that cause it?

**Comment by fadden** [comment]
The plugins sit in a sandbox, but it's in the same app.  It uses the AppDomain feature to keep the code separate, which is exactly what I want, so naturally the feature is deprecated.  There's no network activity involved; the separation is implemented at the language level.  (Which means your security is nonexistent if you can execute "unsafe" code, which is presumably why MSFT gave up on it.)

"Remote" objects time out after 5 minutes by default, with that being reset to 2 minutes on every call.  I override that, but the override doesn't always work, so the Keep-Alive Hack sends a "ping" message to every plugin once every 60 seconds.

The failure suggests that both the lease override *and* the keep-alive hack failed.  Or there's something even more broken about AppDomain security on Windows 8.  Ugh.

It's possible to disable the sandbox entirely, but I made that a compile-time flag.

**Comment by fadden** [comment]
The basic test is: open a project, fiddle with it.  Do something else for 5 minutes.  Change a label (or a code hint, or something that requires re-analyzing the code).  If it crashes 100% of the time, the code in SourceGen is totally wrong.  Without the keep-alive hack it crashes maybe 5% of the time.  With the hack I haven't seen it crash.

I tried the basic test with a release build on Win10 and Win7 (in VirtualBox) and it succeeded, so it's not totally broken.

Was your laptop suspended shortly before the crash?  I'm trying to think of things that could have lead to a timeout.  Playing around just now I determined that I can cause the problem by halting SourceGen in the debugger for 2+ minutes (which is a problem, but not for most people).
